### PR TITLE
Run docsTest as part of the functional test coverage

### PR DESCRIPTION
**Pull request by donat**
We have insufficient coverage for our samples. We should run them through the pipeline just like any other functional tests. This way we can verify that they work on all supported platforms and JVM versions.

Prerequisite for https://github.com/gradle/gradle/pull/13048

### CODE REVIEW

**Comment by blindpirate** [comment]
BTW, I checked `mvn verify` and `./gradlew docs:quickTest`, all looks good.

File: buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/lifecycle/LifecyclePlugin.kt [hunk 1]
```diff
@@ -171,7 +171,7 @@ class LifecyclePlugin : Plugin<Project> {
             description = "Build production distros and smoke test them"
             group = "build"
             dependsOn(":distributions:verifyIsProductionBuildEnvironment", ":distributions:buildDists",
-                ":distributions:integTest", ":docs:check", ":docs:checkSamples")
```
Comment (blindpirate) [inline buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/lifecycle/LifecyclePlugin.kt:174]: Wait a sec, does `docs:check` only depend on `docs:docsTest`? Is there any other tasks `check` depends on?

Comment (donat) [reply buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/lifecycle/LifecyclePlugin.kt:174]: You are right, there are missing tasks. Let me look them up.

Comment (donat) [reply buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/lifecycle/LifecyclePlugin.kt:174]: I've checked the task dependencies and only one task was missing: `releaseNotes`. I've restored it.

File: buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/lifecycle/LifecyclePlugin.kt [hunk 2]
```diff
@@ -184,7 +184,7 @@ class LifecyclePlugin : Plugin<Project> {
             description = "Build production distros, smoke test them and publish"
             group = "publishing"
             dependsOn(":distributions:verifyIsProductionBuildEnvironment", ":distributions:buildDists",
-                ":distributions:integTest", ":docs:check", "publish")
+                ":distributions:integTest", "docs:releaseNotes", "publish")
```
Comment (blindpirate) [inline buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/lifecycle/LifecyclePlugin.kt:187]:
```suggestion
                ":distributions:integTest", ":docs:releaseNotes", "publish")
```

I would suggest keeping the colon...

Comment (donat) [reply buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/lifecycle/LifecyclePlugin.kt]: Fixed!

**Comment by donat** [comment]
Build passed: https://builds.gradle.org/viewLog.html?buildId=34606066

**Pull request merged**